### PR TITLE
[4.x] Workflow Featured

### PR DESCRIPTION
**Pull request by brianteeman**
This pr checks if workflows is enable and if it is changes the "featured status" to not display the Toggle message which is incorrect as it is disabled

Pull Request for Issue #34502 .

### Expected result AFTER applying this Pull Request
image

### CODE REVIEW

**Comment by Kostelano** [comment]
I have tested this item :white_check_mark: successfully on b5d2fa7d026ad53c477af40d39b28deead71d534<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37561">issues.joomla.org/tracker/joomla-cms/37561</a>.</sub>

**Comment by Kostelano** [comment]
Works as it should.

I note that the problem is solved for favorites, but a similar problem for the state of the publication. With the workflow enabled, there is still a line telling you to click to toggle, but there is no clickability in the workflow for these icons.

Screenshot_1

**Comment by brianteeman** [comment]
please create a neew issue for that

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b5d2fa7d026ad53c477af40d39b28deead71d534<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37561">issues.joomla.org/tracker/joomla-cms/37561</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37561">issues.joomla.org/tracker/joomla-cms/37561</a>.</sub>

File: libraries/src/Button/FeaturedButton.php [hunk 1]
```diff
@@ -29,12 +29,22 @@ class FeaturedButton extends ActionButton
 	 */
 	protected function preprocess()
 	{
-		$this->addState(0, 'featured', 'icon-unfeatured',
-			Text::_('JGLOBAL_TOGGLE_FEATURED'), ['tip_title' => Text::_('JUNFEATURED')]
-		);
-		$this->addState(1, 'unfeatured', 'icon-color-featured icon-star',
-			Text::_('JGLOBAL_TOGGLE_FEATURED'), ['tip_title' => Text::_('JFEATURED')]
-		);
+		$workflow_featured = Factory::getApplication()->bootComponent('com_content')->isFunctionalityUsed('core.featured', 'com_content.article');
```
Comment (laoneo) [inline libraries/src/Button/FeaturedButton.php:32]: Is this button only used on com_content? If not, then we shouldn't add here a hard dependency to it.

Comment (brianteeman) [reply libraries/src/Button/FeaturedButton.php:32]: workflow only works on com-content

Comment (laoneo) [reply libraries/src/Button/FeaturedButton.php:32]: Then you need to move the button to com_content. The libraries classes shouldn't have a hard dependency to a specific extension.

Comment (brianteeman) [reply libraries/src/Button/FeaturedButton.php:32]: good luck with that

Comment (laoneo) [reply libraries/src/Button/FeaturedButton.php:32]: Ok, had a look how the button is used. As an instance is directly created in the com_content component you can pass the component reference in the constructor of the button and use that one. If it is not set, do the fallback to com_content.

Comment (richard67) [reply libraries/src/Button/FeaturedButton.php:32]: > workflow only works on com-content

Currently yes, but this is to be changed sooner or later so it works for other components, too.

**Comment by laoneo** [comment]
Back to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37561">issues.joomla.org/tracker/joomla-cms/37561</a>.</sub>

**Comment by laoneo** [comment]
@brianteeman do you have any intention to pass the component to the button?

**Comment by brianteeman** [comment]
no

**Comment by laoneo** [comment]
Ok, then I'm closing this one because of the reasons stated above and will reopen the issue. Thanks for your understanding.

**Comment by brianteeman** [comment]
in another year perhaps someone will look at it again

**Pull request closed without merging**